apps/test/util/testUtils.js

```diff
@@ -303,6 +303,47 @@ export function throwOnConsoleErrorsEverywhere() {
 }
 
 /**
+ * Same as throwOnConsoleErrorsEverywhere, but for warnings
+ */
+let throwingOnWarnings = false;
+export function throwOnConsoleWarningsEverywhere() {
```

Comment:
Seems like we could almost entirely deduplicate this with `throwOnConsoleErrorsEverywhere`, which would be nice since it's a fairly complex bit of logic.  I think it's worth it.  Thoughts?